### PR TITLE
Generic media

### PR DESCRIPTION
**Pull request by edpaget**

### CODE REVIEW

**Comment by edpaget** [comment]
This PR also starts to address #425 by adding an adapter for different storage types besides AWS. 

We'll need a replace our aws.yml config file in production/staging with a storage.yml:

```yaml
staging: 
  prefix: panoptes-uploads.zooniverse.org
  bucket: zooniverse-static
```

**Comment by edpaget** [comment]
Also this is going to break recetns until I finish #696 which I'll make my next priority.

File: lib/subject_locations_extractor.rb [hunk 1]
```diff
@@ -42,13 +42,7 @@ def panoptes_locations
...
   def s3_url(path, mime_type)
     if @context[:post_urls]
-      obj = ::Panoptes.subjects_bucket.objects[path]
-      obj.url_for(:write, {secure: true,
-                           content_type: mime_type,
-                           expires_in: 20.minutes.from_now,
-                           response_content_type: mime_type,
-                           acl: 'public-read'}).to_s
-    else
+   else
```
Comment (camallen) [inline lib/subject_locations_extractor.rb]: should this be `unless @context[]` ?

Comment (edpaget) [inline lib/subject_locations_extractor.rb]: This class isn't being used any more and should be deleted.

File: spec/factories/subjects.rb [hunk 2]
```diff
@@ -21,9 +24,9 @@
       end
     end
 
-    factory :migrated_project_subject do
-      locations({standard: "http://www.galaxyzoo.org.s3.amazonaws.com/subjects/standard/1237679543502373086.jpg"})
-      migrated true
-    end
+    #factory :migrated_project_subject do
```
Comment (camallen) [inline spec/factories/subjects.rb]: prob just remove this if we aren't supporting them

File: db/migrate/20150506195759_remove_avatar_background_from_project.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class RemoveAvatarBackgroundFromProject < ActiveRecord::Migration
+  def change
+    remove_column :projects, :avatar, :text
+    remove_column :projects, :background_image, :text
```
Comment (camallen) [inline db/migrate/20150506195759_remove_avatar_background_from_project.rb]: Are we migrating project avatars and background images?

File: db/migrate/20150506195817_remove_avatar_from_user.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RemoveAvatarFromUser < ActiveRecord::Migration
+  def change
+    remove_column :users, :avatar, :text
```
Comment (camallen) [inline db/migrate/20150506195817_remove_avatar_from_user.rb]: Same, are we migrating these as well?

**Comment by camallen** [comment]
@edpaget - This look good to me, really impressive work!

File: db/migrate/20150506195759_remove_avatar_background_from_project.rb (hunk 3 above)
Comment (edpaget) [inline db/migrate/20150506195759_remove_avatar_background_from_project.rb]: Only the test stargazing projects have them. I figured it'd just be easier to reupload them.

File: db/migrate/20150506195817_remove_avatar_from_user.rb (hunk 4 above)
Comment (edpaget) [inline db/migrate/20150506195817_remove_avatar_from_user.rb]: No one has them yet.

File: spec/factories/subjects.rb (hunk 2 above)
Comment (edpaget) [inline spec/factories/subjects.rb]: We can support them with the external_url in Medium, but they don't require and special handling or tests from the subject model.

File: db/migrate/20150506195759_remove_avatar_background_from_project.rb (hunk 3 above)
Comment (camallen) [inline db/migrate/20150506195759_remove_avatar_background_from_project.rb]: http://demo.zooniverse.org/panoptes-front-end/#/projects/mkosmala/Season%20Spotter%20DETAILS%20Beta1

Comment (edpaget) [inline db/migrate/20150506195759_remove_avatar_background_from_project.rb]: Ah Cool I'll add a migration treating these as externally linked.

**Pull request merged**